crates/sol-type-parser/src/input.rs

```diff
@@ -24,6 +24,8 @@ impl fmt::Display for CustomError {
 
 #[cfg(feature = "std")]
 impl std::error::Error for CustomError {}
+#[cfg(not(feature = "std"))]
+impl core::error::Error for CustomError {}
```

Comment:
this should be no_std already?

this isn't supported yet with the current msrv

Reply:
mb add https://crates.io/crates/rustversion macro here?